sklearn/model_selection/_validation.py

```diff
@@ -474,6 +476,18 @@ def _fit_and_predict(estimator, X, y, train, test, verbose, fit_params,
         estimator.fit(X_train, y_train, **fit_params)
     func = getattr(estimator, method)
     predictions = func(X_test)
+    if method in ['decision_function', 'predict_proba', 'predict_log_proba']:
```

Comment:
I wonder if we should be inspecting the shape of the prediction, rather than doing this based on name.